v3/newrelic/sqlparse/sqlparse.go

```diff
@@ -4,47 +4,168 @@
 package sqlparse
 
 import (
-	"regexp"
+	"bytes"
+	"fmt"
 	"strings"
+	"unicode"
 
 	newrelic "github.com/newrelic/go-agent/v3/newrelic"
 )
 
+var (
+	operations = map[string]string{
+		"select":   "from",
+		"delete":   "from",
+		"insert":   "into",
+		"update":   "",
+		"call":     "",
+		"create":   "",
+		"drop":     "",
+		"show":     "",
+		"set":      "",
+		"exec":     "",
+		"execute":  "",
+		"alter":    "",
+		"commit":   "",
+		"rollback": "",
+	}
+	updateModifiers = map[string]interface{}{
+		"low_priority": nil,
+		"ignore":       nil,
+		"or":           nil,
+		"rollback":     nil,
+		"abort":        nil,
+		"replace":      nil,
+		"fail":         nil,
+		"only":         nil,
+	}
+)
+
+// Extracts the table name from the given string.
 func extractTable(s string) string {
-	s = extractTableRegex.ReplaceAllString(s, "")
 	if idx := strings.Index(s, "."); idx > 0 {
 		s = s[idx+1:]
 	}
-	return s
+
+	var buffer bytes.Buffer
+
+	for _, c := range s {
+		if unicode.IsSpace(c) || strings.ContainsRune("`'(){}[]\"", c) {
```

Comment:
```suggestion
		if unicode.IsSpace(c) || strings.ContainsRune("`'(){}[]\";", c) {
```

Reply:
if you do `SELECT a,b FROM table;` you get the collection `table;`

Reply:
Great catch, thanks for the fix. I also added a test for that case.